File: android/src/main/java/com/visioncameramlkit/domain/models/OutputOrientation.kt
package com.visioncameramlkit.domain.models

/**
 * Dummy enum for iOS parity. Android handles rotation automatically, so this is not used.
 * @see com.visioncameramlkit.domain.models.OutputOrientation (iOS)
 */
enum class OutputOrientation {
  PORTRAIT,
  PORTRAIT_UPSIDE_DOWN,
  LANDSCAPE_LEFT,
  LANDSCAPE_RIGHT,
  ;

  companion object {
    fun fromString(value: String?): OutputOrientation? =
      when (value) {
        "portrait" -> PORTRAIT
        "portrait-upside-down" -> PORTRAIT_UPSIDE_DOWN
        "landscape-left" -> LANDSCAPE_LEFT
        "landscape-right" -> LANDSCAPE_RIGHT
        else -> null
      }
  }
}

File: ios/Domain/Models/OutputOrientation.swift
import UIKit

@objc public enum OutputOrientation: Int {
  case portrait
  case portraitUpsideDown
  case landscapeLeft
  case landscapeRight

  public init?(string: String?) {
    switch string {
    case "portrait": self = .portrait
    case "portrait-upside-down": self = .portraitUpsideDown
    case "landscape-left": self = .landscapeLeft
    case "landscape-right": self = .landscapeRight
    default: return nil
    }
  }

  public var asUIImageOrientation: UIImage.Orientation {
    switch self {
    case .portrait: return .right
    case .portraitUpsideDown: return .left
    case .landscapeLeft: return .down
    case .landscapeRight: return .up
    }
  }
}
